hwconf/Zerno/hw_zerno_drive_core.c

```diff
@@ -36,727 +36,816 @@
+                       PAL_STM32_OSPEED_MID1 |
+                       PAL_STM32_PUDR_PULLUP );
+
+        i2cStart( &HW_I2C_DEV, &i2cfg );
+        i2c_running = true;
+    }
+
+    i2cReleaseBus( &HW_I2C_DEV );
+}
 
-	if (i2c_running) {
-		palSetPadMode(HW_I2C_SCL_PORT, HW_I2C_SCL_PIN, PAL_MODE_INPUT);
-		palSetPadMode(HW_I2C_SDA_PORT, HW_I2C_SDA_PIN, PAL_MODE_INPUT);
+void hw_stop_i2c( void )
+{
+    i2cAcquireBus( &HW_I2C_DEV );
 
-		i2cStop(&HW_I2C_DEV);
-		i2c_running = false;
+    if( i2c_running )
+    {
+        palSetPadMode( HW_I2C_SCL_PORT, HW_I2C_SCL_PIN, PAL_MODE_INPUT );
+        palSetPadMode( HW_I2C_SDA_PORT, HW_I2C_SDA_PIN, PAL_MODE_INPUT );
 
-	}
+        i2cStop( &HW_I2C_DEV );
+        i2c_running = false;
+    }
 
-	i2cReleaseBus(&HW_I2C_DEV);
+    i2cReleaseBus( &HW_I2C_DEV );
 }
 
 /**
  * Try to restore the i2c bus
  */
-void hw_try_restore_i2c(void) {
-	if (i2c_running) {
-		i2cAcquireBus(&HW_I2C_DEV);
-
-		palSetPadMode(HW_I2C_SCL_PORT, HW_I2C_SCL_PIN,
-				PAL_STM32_OTYPE_OPENDRAIN |
-				PAL_STM32_OSPEED_MID1 |
-				PAL_STM32_PUDR_PULLUP);
-
-		palSetPadMode(HW_I2C_SDA_PORT, HW_I2C_SDA_PIN,
-				PAL_STM32_OTYPE_OPENDRAIN |
-				PAL_STM32_OSPEED_MID1 |
-				PAL_STM32_PUDR_PULLUP);
-
-		palSetPad(HW_I2C_SCL_PORT, HW_I2C_SCL_PIN);
-		palSetPad(HW_I2C_SDA_PORT, HW_I2C_SDA_PIN);
-
-		chThdSleep(1);
-
-		for(int i = 0;i < 16;i++) {
-			palClearPad(HW_I2C_SCL_PORT, HW_I2C_SCL_PIN);
-			chThdSleep(1);
-			palSetPad(HW_I2C_SCL_PORT, HW_I2C_SCL_PIN);
-			chThdSleep(1);
-		}
-
-		// Generate start then stop condition
-		palClearPad(HW_I2C_SDA_PORT, HW_I2C_SDA_PIN);
-		chThdSleep(1);
-		palClearPad(HW_I2C_SCL_PORT, HW_I2C_SCL_PIN);
-		chThdSleep(1);
-		palSetPad(HW_I2C_SCL_PORT, HW_I2C_SCL_PIN);
-		chThdSleep(1);
-		palSetPad(HW_I2C_SDA_PORT, HW_I2C_SDA_PIN);
-
-		palSetPadMode(HW_I2C_SCL_PORT, HW_I2C_SCL_PIN,
-				PAL_MODE_ALTERNATE(HW_I2C_GPIO_AF) |
-				PAL_STM32_OTYPE_OPENDRAIN |
-				PAL_STM32_OSPEED_MID1 |
-				PAL_STM32_PUDR_PULLUP);
-
-		palSetPadMode(HW_I2C_SDA_PORT, HW_I2C_SDA_PIN,
-				PAL_MODE_ALTERNATE(HW_I2C_GPIO_AF) |
-				PAL_STM32_OTYPE_OPENDRAIN |
-				PAL_STM32_OSPEED_MID1 |
-				PAL_STM32_PUDR_PULLUP);
-
-		HW_I2C_DEV.state = I2C_STOP;
-		i2cStart(&HW_I2C_DEV, &i2cfg);
-
-		i2cReleaseBus(&HW_I2C_DEV);
-	}
+void hw_try_restore_i2c( void )
+{
+    if( i2c_running )
+    {
+        i2cAcquireBus( &HW_I2C_DEV );
+
+        palSetPadMode( HW_I2C_SCL_PORT, HW_I2C_SCL_PIN,
+                       PAL_STM32_OTYPE_OPENDRAIN |
+                       PAL_STM32_OSPEED_MID1 |
+                       PAL_STM32_PUDR_PULLUP );
+
+        palSetPadMode( HW_I2C_SDA_PORT, HW_I2C_SDA_PIN,
+                       PAL_STM32_OTYPE_OPENDRAIN |
+                       PAL_STM32_OSPEED_MID1 |
+                       PAL_STM32_PUDR_PULLUP );
+
+        palSetPad( HW_I2C_SCL_PORT, HW_I2C_SCL_PIN );
+        palSetPad( HW_I2C_SDA_PORT, HW_I2C_SDA_PIN );
+
+        chThdSleep( 1 );
+
+        for( int i = 0; i < 16; i++ )
```

Comment:
Always try to reduce to the minimum iteration. When using `int`, it encompasses negatives number. Does this for loop needs that range? If not, use `uint_X`. Does the this for loop really needs to be 16 bits? 

```suggestion
        for( uint8_t i = 0; i < 16; i++ )
```